.github/workflows/tests.yml

```diff
@@ -0,0 +1,67 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+
+name: tests 
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+    paths-ignore:
+      - 'docs/**'
+
+jobs:
+  build_and_test:
+    name: Build and test
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres
+        env:
+          POSTGRES_USER: complaint_tracker
+          POSTGRES_DB: complaint_tracker_test
+          POSTGRES_PASSWORD: postgres
+        ports: 
+          - 5432:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+
+    steps:
+      - uses: actions/checkout@v2
+      
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+      
+      - name: Install dependencies
+        run: bundle install
+
+      - name: Setup test database
+        env:
+          RAILS_ENV: test
+          POSTGRES_HOST: localhost
+          POSTGRES_DB: complaint_tracker_test
+          POSTGRES_USER: complaint_tracker 
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_PORT: ${{ job.services.postgres.ports[5432] }}
+        run: |
+          cp config/database.ci.yml config/database.yml
+          bundle exec rake db:create db:migrate
+
+      - name: Precompile assets
+        run: bundle exec rake assets:precompile
+      
+      - name: Run tests
+        env:
+          RAILS_ENV: test
+          POSTGRES_HOST: localhost
+          POSTGRES_DB: complaint_tracker_test
+          POSTGRES_USER: complaint_tracker 
+          POSTGRES_PASSWORD: postgres
+          POSTGRES_PORT: ${{ job.services.postgres.ports[5432] }} 
```

Comment:
Can we build and pass a `DATABASE_URL` env variable here (and in the create/migrate step too)? `DATABASE_URL` always overrides whatever is in `database.yml`, so we wouldn't need to have a separate ci file that needs to be copied over.

Reply:
sure. is there any documentation you can point me to re: DATABASE_URL? i didn't come across it when i was trying to figure this out!

Reply:
https://edgeguides.rubyonrails.org/configuring.html#configuring-a-database section 3.19 and 3.20 go into it.

Reply:
thanks!!